internal/provider/resource_gitlab_pages_domain.go

```diff
@@ -0,0 +1,220 @@
+		Domain:         &domain,
+		AutoSslEnabled: &auto_ssl_enabled,
+		Certificate:    &certificate,
+		Key:            &key,
+	}
+
+	log.Printf("[DEBUG] create gitlab pages domain %s", domain)
+
+	_, _, err := client.PagesDomains.CreatePagesDomain(projectId, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(buildTwoPartID(&projectId, &domain))
+	return resourceGitlabPagesDomainRead(ctx, d, meta)
+}
+
+func resourceGitlabPagesDomainUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	projectID := d.Get("project").(string)
+	domain := d.Get("domain").(string)
+	auto_ssl_enabled := d.Get("auto_ssl_enabled").(bool)
+	certificate := d.Get("certificate").(string)
+	key := d.Get("key").(string)
+
+	options := &gitlab.UpdatePagesDomainOptions{
+		AutoSslEnabled: &auto_ssl_enabled,
+		Certificate:    &certificate,
+		Key:            &key,
+	}
+	log.Printf("[DEBUG] update gitlab pages domain %s for %s", domain, projectID)
+
+	_, _, err := client.PagesDomains.UpdatePagesDomain(projectID, domain, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return resourceGitlabProjectMirrorRead(ctx, d, meta)
+}
+
+func resourceGitlabPagesDomainRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	projectID, domain, err := parseTwoPartID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[DEBUG] read gitlab pages domain %s", domain)
+
+	pagesDomain, _, err := client.PagesDomains.GetPagesDomain(projectID, domain, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			log.Printf("[DEBUG] gitlab pages domain %s not found, removing from state", domain)
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	d.Set("project", projectID)
+	d.Set("domain", pagesDomain.Domain)
+	d.Set("url", pagesDomain.URL)
+	d.Set("auto_ssl_enabled", pagesDomain.AutoSslEnabled)
+	if err := d.Set("certificate_data", flattenCertificateData(pagesDomain)); err != nil {
+		return diag.FromErr(err)
+	}
+	d.Set("verified", pagesDomain.Verified)
+	d.Set("verification_code", pagesDomain.VerificationCode)
+	return nil
+}
+
+func resourceGitlabPagesDomainDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	projectID, domain, err := parseTwoPartID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[DEBUG] Delete gitlab pages domain %s", domain)
+
+	_, err = client.PagesDomains.DeletePagesDomain(projectID, domain, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func flattenCertificateData(pagesDomain *gitlab.PagesDomain) (certificate_data []map[string]interface{}) {
+	if pagesDomain == nil {
+		return
+	}
+
+	var certificate_expiration string
+	if pagesDomain.Certificate.Expiration == nil {
```

Comment:
Can `pagesDomain.Certificate` be `nil` ?

Reply:
No. I tried that first, but it's always initialized with `Expired: false, Expiration: nil`